File: src/main/java/pl/khuzzuk/wfrp/helper/model/crafting/inventory/ArmorPattern.java
package pl.khuzzuk.wfrp.helper.model.crafting.inventory;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Type;
import org.hibernate.validator.constraints.Length;
import pl.javahello.RemoteEntity;
import pl.javahello.RemoteEntity.SecuredService;
import pl.khuzzuk.wfrp.helper.common.EnumType;
import pl.khuzzuk.wfrp.helper.repo.BaseEntity;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

@Getter
@Setter
@Entity
@Table(schema = "crafting")
@RemoteEntity(transactional = true, stomp = true)
@SecuredService(allowRead = true)
public class ArmorPattern extends BaseEntity {
    private @Length(min = 3, max = 100) String name;
    private @Length(max = 500) String description;
    private float priceMultiplier = 1;
    private float weight;
    private float strength;
    @Type(type = EnumType.DEF)
    private @NotNull Availability availability = Availability.COMMON;
}

File: src/wfrp-helper-web/src/model/crafting/ArmorPattern.js
import Entity, {CreateColumn, CreateFormField, RegisterEntity} from "../Entity";
import {Availability, availabilityEntityName} from "./Availability";
import {FormFieldType} from "../../form/FormFieldType";

export default class ArmorPattern extends Entity {
    static entityName: string = 'armorPattern';

    name: string;
    description: string;
    priceMultiplier: number;
    weight: number;
    strength: number;
    availability: string = Availability[0];
}

RegisterEntity(ArmorPattern, [
    CreateColumn('name', FormFieldType.TEXT),
    CreateColumn('description', FormFieldType.TEXT_AREA),
    CreateColumn('strength', FormFieldType.INTEGER),
    CreateColumn('availability', FormFieldType.TEXT),
], [
    CreateFormField('name', FormFieldType.TEXT),
    CreateFormField('description', FormFieldType.TEXT_AREA),
    CreateFormField('priceMultiplier', FormFieldType.FLOAT),
    CreateFormField('weight', FormFieldType.FLOAT),
    CreateFormField('strength', FormFieldType.FLOAT),
    CreateFormField('availability', FormFieldType.ENUM_SELECT, availabilityEntityName),
]);